documentation/sdkautomation/sdk_customization.md

```diff
@@ -0,0 +1,762 @@
+        // Everything in stderr will show in comment and mark package with warning.
+        "showInComment": true,
+        "scriptWarning": true
+      },
+      "exitCode": {
+        // If exit code is not zero, mark package with warning instead of error.
+        "result": "warning"
+      }
+    },
+    
+    // Param: <path_to_package_folder>
+    "changelogScript": {
+      "path": "./eng/tools/sdk_breaking_change",
+      "breakingChangeDetect": "Breaking Change"
+    }
+  },
+  "artifactOptions": {
+    // Param: <path_to_install_instructionInput.json> <path_to_installInstructionOutput.md> <lite|full>
+    // installInstructionInput.json: See #InstallInstructionScriptInput .
+    // installInstructionOutput.md: Install instruction in markdown.
+    "installInstructionScript": {
+      "path": "./eng/tools/sdk_install_instruction"
+    }
+  }
+}
+```
+
+#### SwaggerToSdkConfig Schema
+``` jsonc
+{
+  "type": "object",
+  "properties": {
+    "advancedOptions": {
+      // To keep backward compatibility, but will not list schema for old config options.
+      "properties": {
+        "createSdkPullRequests": {
+          // Should SDK Automation create PR or not.
+          "type": "boolean",
+          "default": true
+        },
+        "closeIntegrationPR": {
+          // Should SDK Automation close integrationPR to reduce noise.
+          "type": "boolean",
+          "default": true
+        },
+        "generationCallMode": {
+          // If we have multiple related readme.md, should we call generation once with
+          // all the readme.md or should we call generation multiple times and one per readme.md.
+          "type": "string",
+          "enum": [
+            "one-per-config",
+            "one-for-all-configs"
+          ],
+          "default": "one-for-all-configs"
+        }
+      }
+    },
+    "initOptions": {
+      // Init the environment. Install dependencies.
+      "type": "object",
+      "properties": {
+        "initScript": {
+          // Script to init.
+          "$ref": "#/definitions/RunOptions"
+        }
+      },
+      "required": [
+        "initScript"
+      ]
+    },
+    "generateOptions": {
+      // Generate the SDK code.
+      "type": "object",
+      "properties": {
+        "generateScript": {
+          // Script to generate the SDK code.
+          // Param: <path_to_generateInput.json> <path_to_generateOutput.json>
+          // generateInput.json: See #GenerateInput
+          // generateOutput.json: See #GenerateOutput
+          "$ref": "#/definitions/RunOptions"
+        },
+        "parseGenerateOutput": {
+          // Will this script output to generateOutput.json.
+          // If not, default behavior will be applied that outcome will be
+          // detected automatically based on filename regex search.
+          "type": "boolean",
+          "default": true
+        }
+      },
+      "required": [
+        "generateScript"
+      ]
+    },
+    "packageOptions": {
```

Comment:
As we discussed from an implementation priority prospective I'd like to get everything up through the genreateScript options. If we can make conversion from existing config files easier without adding all the extra complexity these extra steps we should try to avoid those. The simpler we can make the automation the better. If we can solve some of the complexity with some shared code that would be better then having these extra round tripping and potential error modes between the language repo and the automation.

Reply:
Agree. Will implement generateOption first.